File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = float('inf')   # 目前看過的最低股價
        maxProfit = 0          # 最大利潤
        
        for price in prices:
            if price < minPrice:
                minPrice = price
            elif price - minPrice > maxProfit:
                maxProfit = price - minPrice
        return maxProfit


























        for price in prices:
            # 如果發現更低的價格，就更新 min_price
            if price < min_price:
                min_price = price
            # 否則計算當天賣出的利潤
            elif price - min_price > max_profit:
                max_profit = price - min_price

        return max_profit

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        // 維持buy -> sell -> buy -> sell
        int profit = 0;
        for (int i = 1; i < prices.length; i++)
        {
            if(prices[i] > prices[i - 1])
            {
                profit += (prices[i] - prices[i - 1]);
            }
        }
        return profit;

    }
}